test/TestSubdomainMigrationRegistrar.js

```diff
@@ -0,0 +1,61 @@
+var ENS = artifacts.require("ENSRegistry");
+var SubdomainRegistrar = artifacts.require("SubdomainRegistrar");
+var EthRegistrarSubdomainRegistrar = artifacts.require("EthRegistrarSubdomainRegistrar");
+var SubdomainMigrationRegistrar = artifacts.require("SubdomainMigrationRegistrar");
+var EthRegistrar = artifacts.require("BaseRegistrarImplementation");
+var DummyHashRegistrar = artifacts.require("DummyHashRegistrar");
+var TestResolver = artifacts.require("TestResolver");
+
+var sha3 = require('js-sha3').keccak_256;
+
+contract('SubdomainRegistrar', function (accounts) {
+    var ens = null;
+    var dhr = null;
+    var oldRegistrar = null;
+    var resolver = null;
+    var ethregistrar = null;
+
+    before(async function () {
+        oldRegistrar = await SubdomainRegistrar.deployed();
+        ens = await ENS.deployed();
+        dhr = await DummyHashRegistrar.deployed();
```

Comment:
I'm a little concerned this stub implementation won't replicate the migration behaviour of the real registrar. Any chance you could swap in the real registrar here? See the ethregistrar tests for an example of this.